freqtrade/freqtradebot.py

```diff
@@ -102,6 +106,22 @@ def __init__(self, config: Dict[str, Any]) -> None:
         self._sell_lock = Lock()
         LoggingMixin.__init__(self, logger, timeframe_to_seconds(self.strategy.timeframe))
 
+        if self.config.get("trading_mode"):
+            self.trading_mode = TradingMode(self.config.get("trading_mode"))
+
+        if self.config.get('collateral'):
+            self.collateral = Collateral(self.config.get('collateral'))
+
+        # Start calculating maintenance margin if on cross margin
+        # TODO: Add margin_mode to freqtrade.configuration?
+        if self.collateral == Collateral.CROSS:
+
+            self.maintenance_margin = MaintenanceMargin(
+                exchange_name=self.exchange.name,
+                trading_mode=self.trading_mode)
+
+            self.maintenance_margin.run()
```

Comment:
i'd probably "somehow" bundle this with funding fees (and whatever additional other fees we'll need) - so from a freqtradebot.py perspective, we'll be only calling `handle_running_fees` (naming up to decide) - and that method will then (not in `freqtradebot.py`) decide which method to apply to which trades.